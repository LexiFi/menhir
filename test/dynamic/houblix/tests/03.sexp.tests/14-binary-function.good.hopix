fun : int * int -> int
      fst ((x : int), (y : int)) = x

let main : int = fst (3, 4)