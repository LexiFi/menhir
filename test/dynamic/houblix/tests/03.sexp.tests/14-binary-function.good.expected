(((value
    (DefineValue
      (RecFunctions
        ((((value (Id fst))
            (position ((start_p ("" 2 23 29)) (end_p ("" 2 23 32)))))
           (((value
               (ForallTy ()
                 ((value
                    (TyArrow
                      ((value
                         (TyTuple
                           (((value (TyCon (TCon int) ()))
                              (position
                                ((start_p ("" 1 0 6)) (end_p ("" 1 0 9)))))
                             ((value (TyCon (TCon int) ()))
                               (position
                                 ((start_p ("" 1 0 12)) (end_p ("" 1 0 15))))))))
                        (position ((start_p ("" 1 0 6)) (end_p ("" 1 0 15)))))
                      ((value (TyCon (TCon int) ()))
                        (position
                          ((start_p ("" 1 0 19)) (end_p ("" 1 0 22)))))))
                   (position ((start_p ("" 1 0 6)) (end_p ("" 1 0 22)))))))
              (position ((start_p ("" 1 0 5)) (end_p ("" 1 0 22))))))
           (FunctionDefinition
             ((value
                (PTuple
                  (((value
                      (PTypeAnnotation
                        ((value
                           (PVariable
                             ((value (Id x))
                               (position
                                 ((start_p ("" 2 23 35))
                                   (end_p ("" 2 23 36)))))))
                          (position
                            ((start_p ("" 2 23 35)) (end_p ("" 2 23 36)))))
                        ((value (TyCon (TCon int) ()))
                          (position
                            ((start_p ("" 2 23 39)) (end_p ("" 2 23 42)))))))
                     (position ((start_p ("" 2 23 34)) (end_p ("" 2 23 43)))))
                    ((value
                       (PTypeAnnotation
                         ((value
                            (PVariable
                              ((value (Id y))
                                (position
                                  ((start_p ("" 2 23 46))
                                    (end_p ("" 2 23 47)))))))
                           (position
                             ((start_p ("" 2 23 46)) (end_p ("" 2 23 47)))))
                         ((value (TyCon (TCon int) ()))
                           (position
                             ((start_p ("" 2 23 50)) (end_p ("" 2 23 53)))))))
                      (position
                        ((start_p ("" 2 23 45)) (end_p ("" 2 23 54))))))))
               (position ((start_p ("" 2 23 33)) (end_p ("" 2 23 55)))))
             ((value
                (Variable
                  ((value (Id x))
                    (position ((start_p ("" 2 23 58)) (end_p ("" 2 23 59)))))
                  ()))
               (position ((start_p ("" 2 23 58)) (end_p ("" 2 23 59)))))))))))
   (position ((start_p ("" 1 0 0)) (end_p ("" 2 23 59)))))
  ((value
     (DefineValue
       (SimpleValue
         (((value (Id main))
            (position ((start_p ("" 4 61 65)) (end_p ("" 4 61 69)))))
           (((value
               (ForallTy ()
                 ((value (TyCon (TCon int) ()))
                   (position ((start_p ("" 4 61 72)) (end_p ("" 4 61 75)))))))
              (position ((start_p ("" 4 61 71)) (end_p ("" 4 61 75))))))
           ((value
              (Apply
                ((value
                   (Variable
                     ((value (Id fst))
                       (position
                         ((start_p ("" 4 61 78)) (end_p ("" 4 61 81)))))
                     ()))
                  (position ((start_p ("" 4 61 78)) (end_p ("" 4 61 81)))))
                ((value
                   (Tuple
                     (((value
                         (Literal
                           ((value (LInt 3))
                             (position
                               ((start_p ("" 4 61 83)) (end_p ("" 4 61 84)))))))
                        (position
                          ((start_p ("" 4 61 83)) (end_p ("" 4 61 84)))))
                       ((value
                          (Literal
                            ((value (LInt 4))
                              (position
                                ((start_p ("" 4 61 86)) (end_p ("" 4 61 87)))))))
                         (position
                           ((start_p ("" 4 61 86)) (end_p ("" 4 61 87))))))))
                  (position ((start_p ("" 4 61 82)) (end_p ("" 4 61 88)))))))
             (position ((start_p ("" 4 61 78)) (end_p ("" 4 61 88)))))))))
    (position ((start_p ("" 4 61 61)) (end_p ("" 4 61 88))))))
